final_task/rss_reader/news_parser.py

```diff
@@ -0,0 +1,112 @@
+from bs4 import BeautifulSoup
+import json
+import os
+import urllib.request
+from datetime import datetime
+import time
+import sqlite3
+import  html
+
+
+def checkMediaContent(item):
+    '''check if there is media content'''
+    media_content = ''
+    if ('media_content' in item.keys()):
+        media_content = item.media_content[0]['url']
+    elif ('media_thumbnail' in item.keys()):
+        media_content = item.media_thumbnail[0]['url']
+    return media_content
+
+
+def getDescription(item):
+    '''return description without html tags'''
+    return BeautifulSoup(item, features="html.parser").getText()
+
+
+def intoJson(item):
+    '''print news in json format'''
+    return json.dumps({
+        'Title: ': html.unescape(item.title),
+        'Date: ': item.published,
+        'Link: ': item.link,
+        'Description: ': getDescription(item.description),
+        'Media_link': checkMediaContent(item)
+        })
+
+
+def cacheNews(url, channel):
+    '''
+    1. connect to database
+    2. create table in database
+    3. insert news into table
+    '''
+    conn = sqlite3.connect("newsdatabase.db")
+    cursor = conn.cursor()
+
+    try:
+        cursor.execute("""CREATE TABLE news
+                        (title text, link text, image text, 
+                        description text, pub_date_stamp real,
+                        UNIQUE (title, link, pub_date_stamp))
+                        """)
+    except:
```

Comment:
It is better to not to use bare except clause

Reply:
Fixed